src/Microsoft.ML.StandardLearners/Standard/LinearClassificationTrainer.cs

```diff
@@ -209,7 +209,7 @@ internal virtual void Check(IHostEnvironment env)
                     {
                         ch.Warning("The specified l2Const = {0} is too small. SDCA optimizes the dual objective function. " +
                             "The dual formulation is only valid with a positive L2 regularization. Also, an l2Const less than {1} " +
-                            "could drastically slow down the convergence. So using l2Const = {1} instead.", L2Const);
+                            "could drastically slow down the convergence. So using l2Const = {1} instead.", L2Const, L2LowerBound);
```

Comment:
>l2Const [](start = 83, length = 7)

Hi @justinormont thanks for looking at this. Another problem with this message, replace these with `nameof(L2Const)`, here, and everywhere in this message that it is used, in the format args. It might go easier if you used the `$` string interpolation.